core/src/main/java/com/oath/oak/OakMap.java

```diff
@@ -7,16 +7,25 @@
 package com.oath.oak;
 
 import java.nio.ByteBuffer;
+import java.util.AbstractCollection;
 import java.util.AbstractMap;
+import java.util.AbstractSet;
+import java.util.Collection;
 import java.util.Comparator;
+import java.util.Iterator;
 import java.util.Map;
+import java.util.NavigableSet;
+import java.util.Set;
+import java.util.SortedSet;
+import java.util.concurrent.ConcurrentNavigableMap;
+import java.util.function.BiFunction;
 import java.util.function.Consumer;
 import java.util.function.Function;
 
 /**
  * A concurrent map implementation which supports off-heap memory.
  */
-public class OakMap<K, V> implements AutoCloseable {
+public class OakMap<K, V> extends AbstractMap<K, V> implements AutoCloseable, ConcurrentNavigableMap<K, V> {
```

Comment:
OakMap should not be autoclosable because then you have to call close() for every creation.
This is very confusing and even you in your API test didnt do it:
keySet = oak.subMap(3, 5).keySet(); - the submap function creates a new oak.
You will not be able to do nice things like:
oak.subMap(0, 100).tailMap(2).descendingMap();
this will create 3 calls to a new oakmap that will not be closed ever and the memory will never be free

Reply:
I tend to agree with removing the `AutoCloseable` interface. The current implementation is also not idempotent and double closing can cause problems. 

I suggest we discuss this further and maybe change it in a different pull request.